test/e2e/framework/pod/utils_test.go

```diff
@@ -93,3 +94,57 @@ func TestMixinRestrictedPodSecurity(t *testing.T) {
 	})
 
 }
+
+func TestGetCPULimitCgroupExpectations(t *testing.T) {
+	testCases := []struct {
+		name              string
+		cpuLimit          *resource.Quantity
+		podOnCgroupv2Node bool
+		expected          []string
+	}{
+		{
+			name:              "rounding required, podOnCGroupv2Node=true",
+			cpuLimit:          resource.NewMilliQuantity(15, resource.DecimalSI),
+			podOnCgroupv2Node: true,
+			expected:          []string{"1500 100000", "2000 100000"},
+		},
+		{
+			name:              "rounding not required, podOnCGroupv2Node=true",
+			cpuLimit:          resource.NewMilliQuantity(20, resource.DecimalSI),
+			podOnCgroupv2Node: true,
+			expected:          []string{"2000 100000"},
+		},
+		{
+			name:              "rounding required, podOnCGroupv2Node=false",
+			cpuLimit:          resource.NewMilliQuantity(15, resource.DecimalSI),
+			podOnCgroupv2Node: false,
+			expected:          []string{"1500", "2000"},
+		},
+		{
+			name:              "rounding not required, podOnCGroupv2Node=false",
+			cpuLimit:          resource.NewMilliQuantity(20, resource.DecimalSI),
+			podOnCgroupv2Node: false,
+			expected:          []string{"2000"},
+		},
+		{
+			name:              "cpuQuota=0, podOnCGroupv2Node=true",
+			cpuLimit:          resource.NewMilliQuantity(0, resource.DecimalSI),
+			podOnCgroupv2Node: true,
+			expected:          []string{"max 100000"},
+		},
+		{
+			name:              "cpuQuota=0, podOnCGroupv2Node=false",
+			cpuLimit:          resource.NewMilliQuantity(0, resource.DecimalSI),
+			podOnCgroupv2Node: false,
+			expected:          []string{"-1"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			podOnCgroupv2Node = &tc.podOnCgroupv2Node
```

Comment:
any issues with updating the global var here?

(I didn't see anywhere else it is used besides e2e tests but will be helpful to have confirmation)

Reply:
done!